pkg/controller/controller_test.go

```diff
@@ -3508,6 +3152,16 @@ func TestProvisionFromPVC(t *testing.T) {
 			t.Errorf("test %q: Expected error, got none", k)
 		}
 
+		// Phase: process test responses
+		// reset tc.VolOpts back to defaults
+		if resetSize != nil {
```

Comment:
Instead of having tc carry over across test cases, we should wrap everything around t.Run()

Reply:
Updated the PR adding another commit, that prevents per-tc modification of common content. Also, added the required t.Run() call, and added a t.Parallel() within it (more as proof of each tc not stomping on each others state, rather than for performance).

Assuming this addresses this comment and helps with the cleanup.